File: Data Structures & Algorithms/combination-target-sum-ii/submission-2.java
class Solution {
    public List<List<Integer>> combinationSum2(int[] candidates, int target) {
        Arrays.sort(candidates);
        Set<List<Integer>> res = new HashSet();
        List<Integer> list = new ArrayList();
        backtrack(0, candidates, target, list, res);
        return new ArrayList(res);
    }

    public void backtrack(int i, int candidates[], int target, List<Integer> list, Set<List<Integer>> res){
        if(target==0){
            res.add(new ArrayList(list));
            return;
        }
        if(target<0 || i>=candidates.length) return;
        list.add(candidates[i]);
        backtrack(i+1, candidates, target-candidates[i], list, res);
        list.remove(list.size()-1);
        backtrack(i+1, candidates, target, list, res);
    }
}

File: Data Structures & Algorithms/palindromic-substrings/submission-2.cs
public class Solution {
    public int CountSubstrings(string s) {
        int res = 0;
        for(int i=0;i<s.Length;i++){
            int l = i;
            int r = i;
            while(l>=0 && r<s.Length && s[l]==s[r]){
                res++;
                l--;
                r++;
            }
            l = i;
            r = i+1;
            while(l>=0 && r<s.Length && s[l]==s[r]){
                res++;
                l--;
                r++;
            }
        }
        return res;
    }
}
